plugins/package-managers/nuget/src/main/kotlin/utils/NuGetInspector.kt

```diff
@@ -0,0 +1,299 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.nuget.utils
+
+import com.fasterxml.jackson.module.kotlin.readValue
+
+import java.io.File
+import java.util.SortedSet
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.yamlMapper
+import org.ossreviewtoolkit.utils.common.CommandLineTool
+import org.ossreviewtoolkit.utils.common.collectMessages
+import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.ort.DeclaredLicenseProcessor
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+import org.ossreviewtoolkit.utils.ort.showStackTrace
+
+private val json = Json { ignoreUnknownKeys = true }
+
+internal object NuGetInspector : CommandLineTool {
+    override fun command(workingDir: File?) = "nuget-inspector"
+
+    private fun runNuGetInspector(definitionFile: File, nugetConfig: String?): Result {
+        val workingDir = definitionFile.parentFile
+        val outputFile = createOrtTempFile(prefix = "nuget-inspector", suffix = ".json")
+
+        val commandLineOptions = buildList<String> {
+            add("--with-details")
+            add("--verbose")
+            add("--project-file")
+            add(definitionFile.absolutePath)
+            add("--json")
+            add(outputFile.absolutePath)
+
+            if (nugetConfig != null) {
+                add("--nuget-config")
+                add(workingDir.resolve(nugetConfig).absolutePath)
+            }
+        }
+
+        return runCatching<Result> {
+            try {
```

Comment:
It should be *either* `runCatching` *or* `try`, but not both.

Reply:
So we are pushing this instead:
```Kotlin
        try {
            return runCatching<Result> {
                run(workingDir, *commandLineOptions.toTypedArray())
                outputFile.inputStream().use { json.decodeFromStream(it) }
            }.onFailure { e ->
                e.showStackTrace()
                PackageManager.logger.error {
                    "Unable to determine dependencies for definition file '${definitionFile.absolutePath}': " +
                            e.collectMessages()
                }
                throw e
            }.getOrElse {
                throw it
            }
        } finally {
            workingDir.resolve(".cache").safeDeleteRecursively(force = true)
            outputFile.delete()
        }
```

Would this work for you?